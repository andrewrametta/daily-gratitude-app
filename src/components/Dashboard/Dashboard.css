.ul-days {
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  margin-bottom: 30px;
}

.li-days {
  flex: 1;
  justify-content: space-around;
  font-size: 14px;
  margin-bottom: 20px;
  padding: 15px;
  border-radius: 15px;
  background-color: white;
  margin-left: 15px;
  margin-right: 15px;
  border: none;

  text-indent: -1em;
}

h5 {
  margin-bottom: 15px;
}

h6 {
  color: #177876;
  margin-bottom: 15px;
}

p {
  margin-bottom: 5px;
  padding-left: 10px;
  padding-right: 10px;
}

.date {
  text-align: center;
}

.length {
  text-align: center;
  color: rgb(84, 84, 84);
}

div .button-container-add-new {
  text-align: center;
  margin-bottom: 40px;
}

section .section-dashboard {
  margin-bottom: 30px;
}

@media all and (min-width: 700px) {
  .ul-days {
    margin-right: 25%;
    margin-left: 25%;
  }

  div .button-container-add-new {
    text-align: center;
    margin-top: 15px;
    margin-bottom: 30px;
  }
}
