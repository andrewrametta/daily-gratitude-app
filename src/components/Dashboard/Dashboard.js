import React, { Component } from "react";
import Context from "../../Context";
import { Link } from "react-router-dom";
import "./Dashboard.css";

class Dashboard extends Component {
  static contextType = Context;

  render() {
    // access all days that have been recieved from api and stored into context
    const { days = [] } = this.context;

    return (
      <>
        {days.length === 0 ? <h1>Welcome</h1> : <h1>Your Gratitude Journal</h1>}

        <section>
          {days.length === 0 ? (
            <h5 className="start">Start your gratitude journal today!</h5>
          ) : (
            <h5 className="length">You have {days.length} gratitude entries</h5>
          )}
        </section>
        <section className="section-dashboard">
          {days.length === 0 ? (
            <>
              <div className="button-container-add-new">
                <Link to="/new-day">
                  <button className="button-add-new-entry">
                    Add Your First Entry
                  </button>
                </Link>
              </div>
            </>
          ) : (
            <>
              <ul className="ul-days">
                {days
                  .sort((a, b) => b.id - a.id)
                  .map((day, indx) => (
                    <>
                      <h6 className="date">
                        {new Date(day.date_created).toLocaleDateString()}
                      </h6>
                      <li key={indx} className="li-days">
                        <p>1. {day.text1}</p>
                        <p>2. {day.text2}</p>
                        <p>3. {day.text3}</p>
                      </li>
                    </>
                  ))}
              </ul>
              <div className="button-container-add-new">
                <Link to="/new-day">
                  <button className="button-add-new-entry">
                    Add New Entry
                  </button>
                </Link>
              </div>
            </>
          )}
        </section>
      </>
    );
  }
}

export default Dashboard;
