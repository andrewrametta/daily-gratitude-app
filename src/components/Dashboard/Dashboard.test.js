import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Dashboard from "./Dashboard";
// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe(`Dashboard component`, () => {
  it("renders the complete page", () => {
    const wrapper = shallow(<Dashboard />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
