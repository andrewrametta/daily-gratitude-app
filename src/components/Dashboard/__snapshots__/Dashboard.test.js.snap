// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders the complete page 1`] = `
<div>
  <header>
    <h3>
      Welcome
    </h3>
  </header>
  <section>
    <h5>
      Start your Gratitude Journal Today!
    </h5>
  </section>
  <section>
    <Link
      to="/new-day"
    >
      Add your first day!
    </Link>
  </section>
</div>
`;
